### PR TITLE
fix(cli): quote Windows commandPath when getCommandPath returns null

### PR DESCRIPTION
**Pull request by SleepySML**
## Summary

On Windows, when the user's profile path contains CMD.EXE metacharacters such as `(` (e.g. `C:\Users\Name(Org\...`) and the agent binary is not in PATH so `getCommandPath()` returns `null`, `commandPath` stays as the raw unquoted absolute path from plugin metadata. Passing this to `spawn(..., { shell: true })` routes through `cmd.exe`, which treats bare `(` as a compound-group delimiter and fails with "is not recognized as an internal or external command".

Fixes [EPMCDME-12737](https://jiraeu.epam.com/browse/EPMCDME-12737).

## Changes

- `src/agents/core/BaseAgentAdapter.ts`: quoting guard inserted before `let finalCommand` — wraps `commandPath` in double-quotes when on Windows, path contains CMD.EXE metacharacters (`[ ()&|<>^%[\]{}]`), and path is not already quoted
- `src/agents/core/__tests__/BaseAgentAdapter.test.ts`: three new unit tests covering null-path+metachar (quoted), null-path+clean (unchanged), and non-null-path+metachar (quoted exactly once)

## Testing

- [x] TDD: RED confirmed before guard, GREEN after
- [x] Full unit suite: 2259/2259 tests pass
- [x] Lint: 0 errors, 0 warnings
- [x] Typecheck: 0 diagnostics

## Checklist

- [x] Code follows project standards
- [x] No merge conflicts with `main`

### CODE REVIEW

File: src/agents/core/BaseAgentAdapter.ts [hunk 1]
```diff
@@ -749,6 +749,13 @@ export abstract class BaseAgentAdapter implements AgentAdapter {
         }
       }
 
+      // getCommandPath() may return null (binary not in PATH), leaving commandPath as the raw
+      // unquoted absolute path from plugin metadata. CMD.EXE treats bare '(' as a group delimiter,
+      // so a path like C:\Users\Name(Org\...\bin\cmd.exe must be quoted before shell: true spawn.
+      if (isWindows && /[ ()&|<>^%[\]{}]/.test(commandPath) && !commandPath.startsWith('"')) {
```
Comment (SleepySML) [inline src/agents/core/BaseAgentAdapter.ts]: **CR-001 (major, blocking)** — The character class `/[ ()&|<>^%[\]{}]/` omits cmd.exe **token delimiters** that are legal in Windows directory names: comma, semicolon, equals sign, and tab. cmd.exe splits the command token on all of them, so an unquoted path like `C:\Users\Name;Org\...\cmd.exe` is truncated at the `;` exactly the way the reported path was truncated at `(` — the ticket's "properly escapes all valid Windows directory names" criterion is only partially met.

**Recommendation:** extend the class to `/[ \t,;=()&|<>^%[\]{}]/` — or simplify by quoting any absolute Windows path unconditionally (quoting a benign path is harmless) — and apply the same class to the twin regex at line 713 so the two branches stay consistent.

File: src/agents/core/__tests__/BaseAgentAdapter.test.ts [hunk 2]
```diff
@@ -453,4 +455,83 @@ describe('BaseAgentAdapter', () => {
       }
     });
   });
+
+  describe('run() — Windows command path quoting', () => {
```
Comment (SleepySML) [inline src/agents/core/__tests__/BaseAgentAdapter.test.ts:459]: **CR-002 (major, blocking)** — Jira AC 4 explicitly requires regression tests for "directory name edge cases (**spaces**, parentheses, **etc.**)", but this block exercises only `(` plus a clean path. Spaces — the most common real-world case (`C:\Program Files`) — and every other character in the guard's regex are untested, so a future regex edit could silently drop a character without any test failing.

**Recommendation:** add a parameterized test, e.g. `it.each([' ', '(', ')', '&', '|', '<', '>', '^', '%', '[', ']', '{', '}'])(...)`, asserting the spawned command is quoted. Extend the set alongside CR-001.

File: docs/superpowers/tasks/2026-07-10-epmcdme-12737-windows-special-char-path/qa-report.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# QA Gate Report — epmcdme-12737-windows-special-char-path
```
Comment (SleepySML) [inline docs/superpowers/tasks/2026-07-10-epmcdme-12737-windows-special-char-path/qa-report.md:1]: **CR-003 (major, blocking)** — Jira AC 3 requires "Manual verification in an environment with such a user profile passes as expected", but no manual verification evidence exists in the PR, this qa-report, or the ticket. CI has not run on the branch, and GitHub-hosted runners use clean profile paths anyway — the mocked unit tests are the only evidence, and they cannot prove a real cmd.exe accepts the quoted command line (Node re-wraps it as `cmd /d /s /c "..."`).

**Recommendation:** run `codemie-code` from a directory whose path contains `(` (a temp dir with `(` in its name on any Windows machine approximates the affected profile) and attach the output to the ticket/PR before closing.

**Review by SleepySML — COMMENTED** [review]
## SDLC code-review gate — `code-review.final` → **request-changes** (confidence: high)

Multi-lens review (blind / edge-case / acceptance) for [EPMCDME-12737](https://jiraeu.epam.com/browse/EPMCDME-12737).

**The fix itself is sound** — the quoting guard at `BaseAgentAdapter.ts:755` is correctly placed, matches the approved spec exactly, and solves the reported `(` failure. Three blocking findings survive triage (inline comments below):

| ID | Severity | Summary |
|----|----------|---------|
| CR-001 | major | Guard regex omits cmd.exe token delimiters `,` `;` `=` and tab — same truncation failure persists for those paths |
| CR-002 | major | Jira AC requires regression tests for "spaces, parentheses, etc." — only `(` is exercised |
| CR-003 | major | Jira AC requires manual verification evidence in an affected environment — none attached; CI has not run on the branch |

**Verified and refuted** (checked against branch source at 6ffae59): test mocks *do* exist at module scope (`vi.mock('child_process')` line 92, `getCommandPath` defaults to `Promise.resolve(null)` line 51); `isWindows` is computed inside `run()` (line 706) so the per-test platform spy is effective; the embedded-`"` "injection" scenario is unreachable (`"` is illegal in Windows paths and `cliCommand` is itself the command being executed).

**Deferred (non-blocking follow-ups):**
- `%VAR%` expansion is *not* neutralized by double quotes under cmd.exe; the durable fix is avoiding `shell: true` for absolute `.exe` paths — beyond this ticket's scope.
- `startsWith('"')` is a weak already-quoted sentinel (accepts half-quoted strings) — harden to a balanced-quote check.
- No test covers quoted `commandPath` combined with non-empty args concatenation.
- The quoting regex now drifts across three sites (`BaseAgentAdapter.ts:713`, `:755`, `:760`, plus `exec.ts`) — consolidate into one helper.

_Note: this PR bundles a prior `code-review-final.json` with an `approve` verdict; this review is independent and stricter on the acceptance criteria._

File: src/agents/core/__tests__/BaseAgentAdapter.test.ts (hunk 2 above)
Comment (SleepySML) [reply src/agents/core/__tests__/BaseAgentAdapter.test.ts:459]: ✅ Fixed in 2149cc5 — added a parameterized `it.each` regression test over all 17 metacharacters in the class (space, tab, `,`, `;`, `=`, `()&|<>^%[]{}`). File now has 39 tests; full unit suite 2275 passed / 1 skipped (2276).

File: src/agents/core/BaseAgentAdapter.ts (hunk 1 above)
Comment (SleepySML) [reply src/agents/core/BaseAgentAdapter.ts]: ✅ Fixed in 2149cc5 — character class widened to `/[ \t,;=()&|<>^%[\]{}]/` in both quoting branches (`:713` and `:755`). TDD: the 4 new delimiter cases (tab, `,`, `;`, `=`) failed against the old regex, pass after widening. Real-cmd.exe check: a probe `.cmd` in `...\Name;Org\bin\` failed unquoted with the ticket's exact error and exits 0 with the guard's quoting.

File: docs/superpowers/tasks/2026-07-10-epmcdme-12737-windows-special-char-path/qa-report.md (hunk 3 above)
Comment (SleepySML) [reply docs/superpowers/tasks/2026-07-10-epmcdme-12737-windows-special-char-path/qa-report.md:1]: ✅ Addressed in 6557bfc — manual verification performed on Windows 11 (10.0.26200), Node v22.18.0: a probe `.cmd` spawned via `cmd.exe` (`shell: true`, same call shape as `BaseAgentAdapter.run()`) from `...\Name(Contractor\bin\` reproduces the exact ticket error unquoted (`'C:\Users\...\Name' is not recognized...`, exit 1) and runs clean (exit 0, full path echoed) with the guard's quoting; same result for a `;` path. Evidence table recorded in this qa-report under 'Manual verification'.

**Review by yanaSelin — APPROVED** [review]
